### PR TITLE
Spheretexture

### PR DESCRIPTION
**Pull request by redwoodfavorite**
Fixes sphere texture issues on for first row of vertices.

### CODE REVIEW

File: webgl-geometries/GeometryHelper.js [hunk 1]
```diff
@@ -69,11 +69,13 @@ GeometryHelper.generateParametric = function generateParametric(detailX, detailY
...
 
-    var Xrange = wrap ? Math.PI + (Math.PI / (detailX - 1)) : Math.PI;
+    var offset = (Math.PI / (detailX - 1));
+    var Xrange = wrap ? Math.PI + offset : Math.PI;
+
     var out = [];
 
     for (i = 0; i < detailX + 1; i++) {
-        theta = i * Xrange / detailX;
+        theta = (i === 0 ? 0.0001 : i) * Math.PI / detailX;
```
Comment (michaelobriena) [inline webgl-geometries/GeometryHelper.js]: Can you explain the .0001?

**Comment by redwoodfavorite** [comment]
Okay so I spent a while on this issue, I'll try my best to explain.  The issue is that getting a correct texture coordinate is technically impossible if the coordinates are exactly right.  The problem is that when you try to map a 2D texture onto a 3D sphere you must use the azimuth and altitude of a given vertex to give that vertex a texture coordinate*.  

So the problem arises when you attempt to find the azimuth of the vertex [0, -1, 0] (top vertex) or [0, 1, 0] (bottom vertex).   If a vertex is perfectly centered within a sphere in X and Z there is no logical way to discern the azimuth (progress around the sphere).

Therefore, the solution is to offset the first vertex ever so slightly in the sphere so that logical texture coordinates can be found from the vertex.

And you might be wondering why the bottom vertex does not need to be budged?  This is because Math.sin of Math.PI in javascript does not actually return zero but 1.2246467991473532e-16.

*Azimuth and altitude -> https://upload.wikimedia.org/wikipedia/commons/thumb/f/f7/Azimuth-Altitude_schematic.svg/2000px-Azimuth-Altitude_schematic.svg.png

**Comment by redwoodfavorite** [comment]
This was an issue the Adfab guys pointed out and wanted fixed ASAP btw.

Here is the before and after:

BEFORE: 
https://api-te.famo.us/codemanager/v1/containers/1a99df78-3ed1-43cb-bb47-a58056d9c7bb/share

AFTER: 
https://api-te.famo.us/codemanager/v1/containers/0693fbdc-c4cf-442b-a08e-956f29bc5fb7/share

**Comment by michaelobriena** [comment]
Very interesting.  Nice fix.  Thanks for the detailed explaination.

**Comment by michaelobriena** [comment]
landed as 657db01

**Pull request closed without merging**